library/entropy_poll.c

```diff
@@ -135,10 +140,27 @@ static int has_getrandom = -1;
 
 #include <stdio.h>
 
+#if !defined(HAVE_GETRANDOM)
+static FILE *urandom_file = NULL;
+#endif /* !HAVE_GETRANDOM */
+
+int mbedtls_platform_entropy_init( void )
+{
+#if !defined(HAVE_GETRANDOM)
```

Comment:
`HAVE_GETRANDOM` doesn't guarantee that `getrandom` is available at runtime (we support older Linux kernel versions that some embedded devices run), so the `/dev/urandom` code needs to be included regardless.